File: csharp/Tests/Strata.Diagnostic.CoreTest/Common/DiagnosticCheckTest.cs
using System;
using System.Threading.Tasks;
using Moq;
using NUnit.Framework;
using Strata.Diagnostic.Core.Evaluation;

namespace Strata.Diagnostic.Core.Common
{
    [TestFixture]
    public 
    class DiagnosticCheckTest
    {
	    private DiagnosticCheck target;
	    private static string   NAME = "XYZ";

        [SetUp]
	    public void 
        SetUp()
	    {
		    target = new MockDiagnosticCheck( NAME,true,true,true,false );
	    }

        [TearDown]
	    public void 
        TearDown()
	    {
		    target = null;
	    }

        [Test]
	    public void 
        TestDiagnosticCheck()
	    {
		    Assert.AreEqual( 
		        NAME,
		        new MockDiagnosticCheck( NAME,true,true,true,true ).Name );
	    }

        [Test]
	    public async Task 
        TestRunDiagnostic()
	    {
		    MockDiagnosticResult result   = new MockDiagnosticResult();
		    IDiagnosticReporter  reporter = new Mock<IDiagnosticReporter>().Object;
		
		    result.AttachReporter( reporter );
		    await target.RunDiagnostic( result );
		    Assert.IsTrue( result.ActualRun.Contains( NAME )  );
	    }
    }
}

File: java/Tests/Strata.Diagnostic.CoreTest/src/strata/diagnostic/core/common/DiagnosticCheckTest.java
// ##########################################################################
// # File Name:		DiagnosticCheckTest.java
// # Copyright(C):	2007, Capital Group Companies, Inc. 
// #                All Rights Reserved.
// ##########################################################################

package strata.diagnostic.core.common;

import org.junit.After;
import org.junit.Before;
import org.junit.Test;
import strata.diagnostic.core.evaluation.DiagnosticCheck;

import static org.junit.Assert.assertEquals;
import static strata.foundation.core.utility.Awaiter.await;


/**
 * 
 * @author 		
 *     AFS Strategic Initiative 5 
 * @conventions	
 *     <a href="{@docRoot}/NamingConventions.html">Naming Conventions</a>
 */
public class DiagnosticCheckTest
{
	private DiagnosticCheck itsTarget;
	private static final String NAME = "XYZ";

	/************************************************************************
	 *  
	 *
	 * @throws Exception
	 */
	@Before
	public void setUp() throws Exception
	{
		itsTarget = new MockDiagnosticCheck( NAME,true,true,true,false );
	}

	/************************************************************************
	 *  
	 *
	 * @throws Exception
	 */
	@After
	public void
	tearDown() throws Exception
	{
		itsTarget = null;
	}

	/**
	 * Test method for 
	 * {@link DiagnosticCheck#DiagnosticCheck(String)}.
	 */
	@Test
	public void
	testDiagnosticCheck()
	{
		assertEquals( 
		    NAME,
		    new MockDiagnosticCheck( NAME,true,true,true,true ).getName() );
	}

	/**
	 * Test method for 
	 * {@link DiagnosticCheck#runDiagnostic(IDiagnosticResult)}.
	 */
	@Test
	public void
	testRunDiagnostic()
	{
		MockDiagnosticResult   result   = new MockDiagnosticResult();
		MockDiagnosticReporter reporter = new MockDiagnosticReporter();
		
		result.attachReporter( reporter );
		await(itsTarget.runDiagnostic( result ));
		result.verify();
	}

}


// ##########################################################################
